colibre/registration.py

```diff
@@ -845,10 +845,205 @@ def register_star_Mg_and_O_to_Fe(self, catalogue, aperture_sizes):
+            np.zeros_like(gas_cold_dense_mass), "dimensionless"
+        )
+
+        # Avoid division by zero
+        mask = gas_cold_dense_mass > 0.0 * gas_cold_dense_mass.units
+        log_N_over_O[mask] = np.log10(
+            log_N_over_O_times_gas_mass[mask] / gas_cold_dense_mass[mask]
+        )
+
+        log_N_over_O.name = (
+            f"Diffuse Gas $\\log_{{10}}({{\\rm N/O}})$ ({aperture_size} kpc)"
+        )
+
+        # Register the field
+        setattr(
+            self, f"gas_n_over_o_abundance_avglin_{aperture_size}_kpc", log_N_over_O
+        )
+        setattr(self, f"has_cold_dense_gas_{aperture_size}_kpc", mask)
+
+    # Loop over aperture average-of-log N/O-abundances
+    for aperture_size in aperture_sizes:
+
+        # register average-of-log O-abundances (high and low particle floors)
+        for floor, floor_label in zip(
+            ["low", "high"], ["Min = $10^{{-4}}$", "Min = $10^{{-3}}$"]
+        ):
+            # Fetch N over O times gas mass computed in apertures.
+            # Note that here we are calling the diffuse quantities
+            log_N_over_O_times_gas_mass = catalogue.get_quantity(
+                f"log_element_ratios_times_masses.log_N_over_O_times_gas_mass_{floor}floor_{aperture_size}_kpc"
+            )
+
+            # Fetch gas mass in apertures
+            gas_cold_dense_mass = catalogue.get_quantity(
+                f"cold_dense_gas_properties.cold_dense_gas_mass_{aperture_size}_kpc"
+            )
+
+            # Compute gas-mass weighted N over O
+            log_N_over_O = unyt.unyt_array(
+                np.zeros_like(gas_cold_dense_mass), "dimensionless"
+            )
+            # Avoid division by zero
+            mask = gas_cold_dense_mass > 0.0 * gas_cold_dense_mass.units
+            log_N_over_O[mask] = (
+                log_N_over_O_times_gas_mass[mask] / gas_cold_dense_mass[mask]
+            )
+
+            # Convert to units used in observations
+            N_abundance = unyt.unyt_array(log_N_over_O, "dimensionless")
+            N_abundance.name = f"Diffuse Gas $\\log_{{10}}({{\\rm N/O}})$ ({floor_label}, {aperture_size} kpc)"
+
+            # Register the field
+            setattr(
+                self,
+                f"gas_n_over_o_abundance_avglog_{floor}_{aperture_size}_kpc",
+                N_abundance,
+            )
+            setattr(self, f"has_cold_dense_gas_{aperture_size}_kpc", mask)
+
+    return
+
+
+def register_carbon_to_oxygen(self, catalogue, aperture_sizes):
+    # Loop over aperture average-of-linear C/O-abundances
+    for aperture_size in aperture_sizes:
+
+        # Fetch C over O times gas mass computed in apertures. The
+        # mass ratio between N and O has already been accounted for.
+        # Note that here we are calling the diffuse quantities, for
+        # total quantities (diffuse + dust) call "lin_C_over_O_total_times_ .."
+        log_C_over_O_times_gas_mass = catalogue.get_quantity(
```

Comment:
Is it possible to copy the loop [here](https://github.com/correac/pipeline-configs/blob/8452738affeb21bd79ebb48c254b90512c1a958a/colibre/registration.py#L1047) to register the diffuse and total quantities with a bit less code?

Reply:
Yes! Good point. I included the loop in both the C/O and N/O registration functions.